packages/components/data-table/src/column-resizer.js

```diff
@@ -20,11 +17,13 @@ const ResizerIndicator = styled.div`
 `;
 
 const DraggableArea = styled.div`
+  display: flex;
+  justify-content: flex-end;
   height: 100%;
   position: absolute;
   top: 0;
-  width: ${resizerHoverArea}px;
-  right: -${resizerHoverArea - resizerWidth}px;
```

Comment:
Instead of keeping the draggable area centered on the edge, overlapping the right side for a few pixels, I moved it to stay on the inner edge of the header cell.